### PR TITLE
Add Working Groups to the istio/istio repo, clean up some of the other guidelines

### PR DESCRIPTION
**Pull request by smawson**
This includes the new list of working groups plus some changes to relevant pages to point at the working groups. I've also cleaned up and simplified some of the verbiage on some of the other pages.

### CODE REVIEW

File: wgs.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+| Networking | Andra Cismaru (Google), Kuat Yessenov (Google), Shriram Rajagopalan (IBM), Christopher Luciano (IBM) | [Mailing List](https://groups.google.com/forum/#!forum/istio-networking) | Pilot integration, TCP Support, Additional L7 protocols, Proxy injection |
+| Environments | Costin Manolache (Google), Laurent Demailly (Google), Jose Ortiz (IBM) | [Mailing List](https://groups.google.com/forum/#!forum/istio-environments) | Raw VM support, Hybrid Mesh, Mac/Windows support, Cloud Foundry integration |
+| Integrations | Martin Taillefer (Google), Todd Kaplinger (IBM) | [Mailing List](https://groups.google.com/forum/#!forum/istio-integrations) | Mixer Adapter Model, Rate Limiting, Tracing, Monitoring, Logging |
+| API Manaagement | Wencheng Lu (Google), Jason Allor (Google), Tony Ffrench (IBM) | [Mailing List](https://groups.google.com/forum/#!forum/istio-api-management) | API Keys, Content Mediation, Content Translation, OpenAPI Ingestion |
```
Comment (kyessenov) [inline wgs.md]: typo

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -78,23 +60,12 @@ feature requests.
...
-Before checking in a new feature, you are expected to contribute full
-documentation for this feature within the main Istio repo (https://github.com/istio/istio).
-
-Once the idea has be discussed and there is a general agreement on the
-technical direction, a PR can then be submitted.
+In order to contribute a feature to Istio you'll need to go through the following steps:
+- Discuss your idea with the appropriate [working groups](wgs.md).
+- Once there is general agreement that the feature is useful, create a GitHub issue to track the discussion. The issue should include information about the requirements and use cases that it is trying to address.
+- Include a discussion of the proposed design and technical details of the implementation in the issue.
+- Contribute documentation of the feature to the main Istio repo (https://github.com/istio/istio).
```
Comment (geeknoid) [inline CONTRIBUTING.md]: Docs should go to istio.github.io.

File: wgs.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Working Groups
```
Comment (geeknoid) [inline wgs.md]: Can we expand wgs into a full name such as GROUPS.md (following the all uppercase convention for these top-level docs)

File: CONTRIBUTING.md (hunk 2 above)
Comment (rshriram) [inline CONTRIBUTING.md]: and swap these two steps..

Comment (smawson) [inline CONTRIBUTING.md]: Swap which steps?  You want documentation before the design?

Comment (smawson) [inline CONTRIBUTING.md]: Oh I think you mean documentation after submitting PR?  will do.  Also will swap to istio.github.io.

File: wgs.md (hunk 3 above)
Comment (smawson) [inline wgs.md]: Will do.

File: wgs.md (hunk 1 above)
Comment (smawson) [inline wgs.md]: Fixed.

**Comment by smawson** [comment]
re: istio.io, yeah I'll update that too.

**Pull request merged**